specification/network/resource-manager/Microsoft.Network/stable/2024-05-01/ipAddressManager.json

```diff
@@ -0,0 +1,1119 @@
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "skip",
+            "description": "Optional num entries to skip.",
+            "type": "integer",
+            "format": "int32",
+            "default": 0
+          },
+          {
+            "in": "query",
+            "name": "top",
+            "description": "Optional num entries to show.",
+            "type": "integer",
+            "format": "int32",
+            "default": 50
+          },
+          {
+            "in": "query",
+            "name": "sortKey",
+            "description": "Optional key by which to sort.",
+            "type": "string"
+          },
+          {
+            "in": "query",
+            "name": "sortValue",
+            "description": "Optional sort value for pagination.",
+            "type": "string"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of IpamPools",
+            "schema": {
+              "$ref": "#/definitions/IpamPoolList"
+            }
+          },
+          "default": {
+            "description": "Default error response.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CommonErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "IpamPools_List": {
+            "$ref": "./examples/IpamPools_List.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/networkManagers/{networkManagerName}/ipamPools/{poolName}": {
+      "put": {
+        "tags": [
+          "IpamPools"
+        ],
+        "summary": "Creates/Updates the Pool resource.",
+        "operationId": "IpamPools_Create",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./networkManager.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./networkManager.json#/parameters/NetworkManagerNameParameter"
+          },
+          {
+            "in": "path",
+            "name": "poolName",
+            "description": "IP Address Manager Pool resource name.",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9-]*$"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "in": "body",
+            "name": "body",
+            "description": "Pool resource object to create/update.",
+            "required": true,
```

Comment:
>"required": true, [](http://example.com/codeflow?start=12&length=17)

compared to 18922 from private this was added only in this pr

Reply:
We added this for azure client tools like CLI to generate the right required parameters. Yes, this was after the private PR.